mitodl/open-discussions
# Open drawer for programs

#### Pre-Flight checklist


#### What are the relevant tickets?
Partially addresses #2152

#### What's this PR do?
Opens the `LearningResourceDrawer` when a program search result/card is clicked and displays the relevant info.

#### How should this be manually tested?
- import some programs if you don't have any yet:
  ```
   from course_catalog.tasks import *
   get_xpro_data()
   sync_and_upload_edx_data()
   get_micromasters_data()
  ```
  you'll need to set some .env variables for the above including:
  ```
  MICROMASTERS_CATALOG_API_URL=http://docker.for.mac.host.internal:8079/api/v0/catalog/
  XPRO_CATALOG_API_URL=http://docker.for.mac.host.internal:8053/api/programs/
  ```
- Search for programs at `/courses/search`.  Click one of the results.  The drawer should open with the relevant info and it should look roughly like the [design](https://projects.invisionapp.com/d/main#/console/17477251/376322443/preview), but many fields will be missing due to the info not existing for programs (length, effort, video transcript, topics, instructors, etc).
- ~The image will be broken for xpro courses because the xpro API lists them as relative URL's (will be fixed soon)~.
- If there is a link to a Micromasters program, it won't work either because it is imported as a relative URL (https://github.com/mitodl/micromasters/issues/4410)

#### Screenshots (if appropriate)
<img width="495" alt="Screen Shot 2019-09-27 at 10 35 37 AM" src="https://user-images.githubusercontent.com/187676/65777737-9b240200-e112-11e9-96a9-0f88b3973141.png">


<img width="498" alt="Screen Shot 2019-09-27 at 10 33 52 AM" src="https://user-images.githubusercontent.com/187676/65777636-6912a000-e112-11e9-9069-bbd2da168238.png">
